src/tests/unit_tests/tools/test_search_tool.py

```diff
@@ -10,6 +14,84 @@ def test_search_query_includes_gsite_config():
     )
     assert config.gsite == site
     search_tool = SearchTool(config=config)
+
+    query = "What is the weather today?"
+    search_result = search_tool._run(query)
+    assert search_result is not None
+    assert search_result is not ""
+
+
+def test_search_query_includes_multiple_gsite_config():
```

Comment:
Here, we are testing if the URLs can be parsed correctly. Ideally, we shouldn't need to call the actual Google Search API to test the parsing functionality. We should factor out the parsing of URLs and test it separately. 

If we move the parsing to the configuration. We should move this test to the configuration as well.